mm2src/coins/lp_coins.rs

```diff
@@ -410,6 +430,65 @@ pub struct RawTransactionRes {
     pub tx_hex: BytesJson,
 }
 
+/// Previous utxo transaction data for signing
+#[derive(Clone, Debug, Deserialize)]
+pub struct PrevTxns {
+    /// transaction hash
+    pub tx_hash: String,
+    /// transaction output index
+    pub index: u32,
+    /// transaction output script pub key
+    pub script_pub_key: String,
+    // TODO: implement if needed:
+    // redeem script for P2SH script pubkey
+    // pub redeem_script: Option<String>,
+    /// transaction output amount
+    pub amount: BigDecimal,
+}
+
+/// RPC request with unsigned utxo transaction and params for signing
+#[derive(Clone, Debug, Deserialize)]
+pub struct SignRawTransactionRequest {
+    /// coin ticker name
+    pub coin: String,
+    /// unsigned utxo transaction in hex
+    pub tx_hex: String,
+    /// optional data of previous transactions referred by unsigned transaction inputs
+    pub prev_txns: Option<Vec<PrevTxns>>,
+    // TODO: add if needed:
+    // pub sighash_type: Option<String>, optional signature hash type, one of values: NONE, SINGLE, ALL, NONE|ANYONECANPAY, SINGLE|ANYONECANPAY, ALL|ANYONECANPAY (if not set 'ALL' is used)
+    // pub branch_id: Option<u32>, zcash or komodo optional consensus branch id, used for signing transactions ahead of current height
+}
+
+/// RPC response with signed utxo transaction
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct SignRawTransactionResponse {
+    /// Raw bytes of signed transaction in hexadecimal string which is the response from the sign_raw_transaction request
+    pub tx_hex: BytesJson,
+}
+
+/// RPC request with unsigned eth transaction and params for signing
+#[derive(Clone, Debug, Deserialize)]
+pub struct SignEthTransactionRequest {
+    /// Coin ticker name
+    pub coin: String,
+    /// Eth transfer value
+    value: Option<U256>,
+    /// Eth to address
+    to: Option<String>,
```

Comment:
I think instead of `to` we should have `action` that takes an enum SignAction
```rust
#[derive(Clone, Debug, Deserialize)]
#[serde(tag = "type")]
pub enum SignAction {
    Create,
    Call { address: String },
}
```
So the request can be either
```json
"action": {
   "type": "Create"
}
```
or
```json
"action": {
   "type": "Call",
   "address": "0x7Bc1bBDD6A0a722fC9bffC49c921B685ECB84b94"
}
```
This way the user can specify the specific action desired instead of calling create if they forgot to add an address.

Reply:
I used 'to' to make this rpc similar to the original web3 api signTransaction call. 
But I can change it to 'action' though..

Reply:
we agreed we can leave it as it is now